packages/wallets/ledgerconnect/src/adapter.ts

```diff
@@ -0,0 +1,236 @@
+import {
+    BaseSignerWalletAdapter,
+    EventEmitter,
+    scopePollingDetectionStrategy,
+    SendTransactionOptions,
+    WalletAccountError,
+    WalletConnectionError,
+    WalletDisconnectedError,
+    WalletDisconnectionError,
+    WalletError,
+    WalletName,
+    WalletNotConnectedError,
+    WalletNotReadyError,
+    WalletReadyState,
+    WalletSignTransactionError,
+} from '@solana/wallet-adapter-base';
+import { Connection, PublicKey, SendOptions, Transaction, TransactionSignature } from '@solana/web3.js';
+
+interface LedgerConnectWalletEvents {
+    connect(...args: unknown[]): unknown;
+    disconnect(...args: unknown[]): unknown;
+}
+
+interface LedgerConnectWallet extends EventEmitter<LedgerConnectWalletEvents> {
+    isLedgerConnect?: boolean;
+    publicKey?: PublicKey;
+    isConnected: boolean;
+    signTransaction(transaction: Transaction): Promise<Transaction>;
+    signAndSendTransaction(
+        transaction: Transaction,
+        options?: SendOptions
+    ): Promise<{ signature: TransactionSignature }>;
+    signAllTransactions(transactions: Transaction[]): Promise<Transaction[]>;
+    connect(): Promise<void>;
+    disconnect(): Promise<void>;
+}
+
+interface LedgerConnectWindow extends Window {
+    ledgerConnectSolana?: LedgerConnectWallet;
+}
+
+declare const window: LedgerConnectWindow;
+
+export interface LedgerConnectWalletAdapterConfig {}
+
+export const LedgerConnectWalletName = 'Ledger Connect' as WalletName;
+
+export class LedgerConnectWalletAdapter extends BaseSignerWalletAdapter {
+    name = LedgerConnectWalletName;
+    url = 'https://www.ledger.com';
+    icon =
+        'data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYwIiBoZWlnaHQ9IjE2MCIgdmlld0JveD0iMCAwIDE2MCAxNjAiIGZpbGw9Im5vbmUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CjxyZWN0IHdpZHRoPSIxNjAiIGhlaWdodD0iMTYwIiByeD0iMTYiIGZpbGw9IndoaXRlIi8+CjxwYXRoIGQ9Ik05My4xNDgyIDExOS4yMDdWMTI1SDEzNVY5OC44NzY5SDEyOC45MDJWMTE5LjIwN0g5My4xNDgyWk05My4xNDgyIDMzVjM4Ljc5MkgxMjguOTAyVjU5LjEyMzFIMTM1VjMzSDkzLjE0ODJaTTc0LjAxMDQgNTkuMTIzMUg2Ny45MTI1Vjk4Ljg3NjlIOTUuNDE1M1Y5My42NTM5SDc0LjAxMDRWNTkuMTIzMVpNMjYgOTguODc2OVYxMjVINjcuODUxOFYxMTkuMjA3SDMyLjA5NzlWOTguODc2OUgyNlpNMjYgMzNWNTkuMTIzMUgzMi4wOTc5VjM4Ljc5Mkg2Ny44NTE4VjMzSDI2WiIgZmlsbD0iIzAwMDAwRCIvPgo8L3N2Zz4K';
```

Comment:
Might be a dumb question since I lack the context of where this [ L ] logo is shown but, would it work well also on dark mode for instance?

Reply:
Yep I tested that and it retains the white background in dark mode. Take a look at the demo video I posted in slack for an example of what it looks like.